lib/decorate.js

```diff
@@ -1,44 +1,47 @@
 'use strict'
 
-function decorate (name, fn, dependencies) {
-  if (checkExistence(this, name)) {
+function decorate (instance, name, fn, dependencies) {
+  if (checkExistence(instance, name)) {
     throw new Error(`The decorator '${name}' has been already added!`)
   }
 
   if (dependencies) {
-    checkDependencies(this, dependencies)
+    checkDependencies(instance, dependencies)
   }
 
   if (fn && (typeof fn.getter === 'function' || typeof fn.setter === 'function')) {
-    Object.defineProperty(this, name, {
+    Object.defineProperty(instance.prototype || instance, name, {
```

Comment:
Are you sure about this `||` conditional? It seems to me everything in Fastify that can be decorated will have a prototype.

Reply:
My apologies, that is left over from some previous attempts, shouldn't be needed